### PR TITLE
Make compatible with ARM processors (also armv8).

### PR DESCRIPTION
**Pull request by fabianbl**

### CODE REVIEW

File: aslam_cv_common/cmake/detect_simd.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Detect the preprocessor directives which are set by the compiler.
+execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dM -E -x c /dev/null
+                OUTPUT_VARIABLE PREPROCESSOR_DIRECTIVES)
+
+set(IS_SSE_ENABLED FALSE)
+set(IS_NEON_ENABLED FALSE)
+if (PREPROCESSOR_DIRECTIVES MATCHES "__SSE2__")
```
Comment (mfehr) [inline aslam_cv_common/cmake/detect_simd.cmake]: Is `__SSE2__` actually covering SSE3 instructions as well? SSE3 does provide some additional instructions.

File: aslam_cv_common/include/aslam/common/hamming-inl.h [hunk 2]
```diff
@@ -41,9 +41,9 @@
 #ifndef ASLAM_COMMON_HAMMING_INL_H_
 #define ASLAM_COMMON_HAMMING_INL_H_
 
-#ifdef __ARM_NEON__
```
Comment (mfehr) [inline aslam_cv_common/include/aslam/common/hamming-inl.h]: could you ellaborate on why this change was necessary?

Comment (fabianbl) [reply aslam_cv_common/include/aslam/common/hamming-inl.h]: __ARM_NEON__ is legacy and will only be set for `armv7` processors, whereas __ARM_NEON is set for both `armv7` and `armv8` processors (see [here](http://www.openwall.com/lists/john-dev/2016/01/18/1))

File: aslam_cv_common/cmake/detect_simd.cmake (hunk 1 above)
Comment (fabianbl) [reply aslam_cv_common/cmake/detect_simd.cmake]: @mfehr On current Intel/AMD processors, it seems to be quite likely that SSE3 is supported (see [here](https://stackoverflow.com/a/28186601)), but not strictly necessary. So I can add a check if it is really there.
